editor/editor_log.cpp

```diff
@@ -58,8 +58,8 @@ void EditorLog::_error_handler(void *p_self, const char *p_func, const char *p_f
 
 	MessageType message_type = p_type == ERR_HANDLER_WARNING ? MSG_TYPE_WARNING : MSG_TYPE_ERROR;
 
-	if (self->current != Thread::get_caller_id()) {
-		callable_mp(self, &EditorLog::add_message).call_deferred(err_str, message_type);
+	if (!Thread::is_main_thread()) {
+		MessageQueue::get_main_singleton()->push_callable(callable_mp(self, &EditorLog::add_message).bind(err_str, message_type));
```

Comment:
```suggestion
		MessageQueue::get_main_singleton()->push_callable(callable_mp(self, &EditorLog::add_message), err_str, message_type);
```

Reply:
I keep forgetting about that possibility. 😅 Will push the fix.